topics/tutorials/custom_language_support/structure_aware_navbar.md

```diff
@@ -0,0 +1,37 @@
+[//]: # "title: 15. Structure Aware Navigation Bar"
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+Structure aware navbar allows displaying specific file elements in the [navigation bar](https://www.jetbrains.com/help/idea/guided-tour-around-the-user-interface.html#navigation-bar), depending on the location of the caret in it. For example, in Java this is used to display a class and method in which the caret locates.
+
+**Reference**: [Navigation Bar](navbar.md)
+
+## Define a SimpleStructureAwareNavbar
+
+The simple structure aware navbar implements [`StructureAwareNavBarModelExtension`](upsource:///platform/lang-impl/src/com/intellij/ide/navigationToolbar/StructureAwareNavBarModelExtension.java).
+
+Override the `getLanguage()` method to return the language for which the navigation bar will be used. To define text for a file and a property, override the `getPresentableText()` method, and finally, to display an icon for a property, override the `getIcon()` method.
```

Comment:
IMHO we could remove this paragraph, as it's clear from implementation and linked reference page

Reply:
Sounds reasonable, removed.